ci/custom_cops/lib/custom_cops/assert_not.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module CustomCops
+  # Enforces the use of `assert_not` over `assert !`.
+  #
+  # @example
+  #   # bad
+  #   assert !x
+  #   assert ! x
+  #
+  #   # good
+  #   assert_not x
+  #
+  class AssertNot < RuboCop::Cop::Cop
+    MSG = "Prefer `assert_not` over `assert !`"
+
+    def_node_matcher :offensive?, "(send nil? :assert (send ... :!))"
+
+    def on_send(node)
+      return unless offensive?(node)
+      add_offense(node, location: :expression, message: MSG)
```

Comment:
Small nitpick, you don't need the `location: :expression` as well as the message kwarg. https://github.com/bbatsov/rubocop/blob/72bb3922b071712a9262df04e6f5ab5e6f509088/lib/rubocop/cop/internal_affairs/redundant_location_argument.rb#L12-L16
The msg is automatically inferred from the constant

Reply:
Thanks!